xla/service/latency_hiding_scheduler.h

```diff
@@ -960,8 +961,9 @@ class DefaultSchedulerCore : public SchedulerCore {
     std::vector<HloInstruction*> new_sequence_reversed;
     // Units of time passed in the schedule. To keep track of latency hiding.
     HloGraphNode::TimeCost current_time = 0;
-    // Number of resources in flight.
-    ResourceMap resources_in_flight;
+    // Resources and corresponding occupiers in flight.
+    absl::flat_hash_map<int64_t, absl::node_hash_set<const HloInstruction*>>
```

Comment:
Is there a reason why you prefer `absl::node_hash_set` here (instead of `absl::flat_hash_set`)?

Reply:
it's about pointer stability, those sets are iterated at some places and there could be issues if we pass them by pointers -- this is not happening at the moment but I think using node_hash_set can be less error-prone for potential future usage.

Reply:
I doubt we will ever pass those by pointers. Can you change it to flat_hash_set please (and remove the newly added dependency)? This is probably already incurring some slowdown; so we want to keep it minimal.